packages/roosterjs-editor-plugins/lib/plugins/TableResize/TableResize.ts

```diff
@@ -496,177 +496,187 @@ export default class TableResize implements EditorPlugin {
                             }
                         }
                     }
                 }
             }
-            rect = normalizeRect(this.currentTable.getBoundingClientRect());
-            currentBorder = this.isRTL ? rect.left : rect.right;
-            this.currentTable.setAttribute(
-                this.isRTL ? 'currentLeftBorder' : 'currentRightBorder',
-                currentBorder.toString()
-            );
+        }
+        rect = normalizeRect(vtable.table.getBoundingClientRect());
+        currentBorder = this.isRTL ? rect.left : rect.right;
+        vtable.table.setAttribute(
+            this.isRTL ? 'currentLeftBorder' : 'currentRightBorder',
+            currentBorder.toString()
+        );
 
-            const currentBottomBorder = this.currentTable.getBoundingClientRect().bottom;
-            this.currentTable.setAttribute('currentBottomBorder', currentBottomBorder.toString());
-            vtable.writeBack();
+        const currentBottomBorder = vtable.table.getBoundingClientRect().bottom;
+        vtable.table.setAttribute('currentBottomBorder', currentBottomBorder.toString());
+        vtable.writeBack();
+        return;
+    };
+
+    private resizeRows = (newPos: number, rect: Rect, vtable: VTable) => {
+        if (!this.isReadyForResizingHeight) {
+            preProcessTable(vtable.table, ResizeState.Horizontal);
+            this.isReadyForResizingHeight = true;
+        }
+        vtable.forEachCellOfCurrentRow(cell => {
+            if (cell.td) {
+                cell.td.style.height = cell.td == this.currentTd ? `${newPos - rect.top}px` : null;
+            }
+        });
+    };
+
+    private resizeColumns = (newPos: number, vtable: VTable, isShiftPressed: boolean) => {
+        if (!this.isReadyForResizingWidth) {
+            preProcessTable(vtable.table, ResizeState.Vertical);
+            this.isReadyForResizingWidth = true;
+        }
+        // Since we allow the user to resize the table width on adjusting the border of the last cell,
+        // we need to make the table width resizeable by setting it as null;
+        // We also allow the user to resize the table width if Shift key is pressed
+        if (!this.currentCellsToResize[0].nextElementSibling || isShiftPressed) {
+            vtable.table.removeAttribute('width');
+            vtable.table.style.width = null;
+        }
+
+        this.currentCellsToResize.forEach(td => {
+            const rect = normalizeRect(td.getBoundingClientRect());
+            const nextTd = td.nextElementSibling as HTMLTableElement;
+            td.style.wordBreak = 'break-word';
+            td.style.whiteSpace = 'normal';
+            td.style.boxSizing = 'border-box';
+
+            const nextTdWidth = nextTd
+                ? this.isRTL
+                    ? newPos - normalizeRect(nextTd.getBoundingClientRect()).left
+                    : normalizeRect(nextTd.getBoundingClientRect()).right - newPos
```

Comment:
This pattern occurs multiple times. Suggest extract to a function:

function getDistance(rect: Rect, pos: number, isRTL: boolean, toLeft: boolean) {
  return !!isRTL == !!toLeft ? newPos - rect.left : rect.right - newPos
}

Reply:
hmm...I can see there is only one pattern here, where are other ones?

Reply:
Line 550, Line 649, Line 656, Line 663

Reply:
I feel like the current pattern looks more straightforward to me, having a function with 4 parameter feels more obscure to understand, let's keep it as it is?

Reply:
The goal is to reduce duplicated code. In your code here, normalizeRect(nextTd.getBoundingClientRect()) appears twice.

Reply:
fixed